### ISSUE TITLE
update for Enphase API v4?

### ISSUE THREAD

**Issue by davepacheco** [NathanBaulch/EnphaseOpenAPI#2]
First, thanks for this!  I've been using the OpenAPI spec to generate a Rust client for monitoring my production for the last year.

I got a message from Enphase today that they're [EOL'ing the v2 API](https://developer-v4.enphase.com/aboutproduct.html) (which is what the OpenAPI spec says it describes) in January.  Any chance you'd consider updating this to the v4 API?  There's a [summary of changes here](https://developer-v4.enphase.com/docs/release_notes).

**Comment by NathanBaulch** [comment]
Yeah, I got the same message too. I guess if there's interest in it then I can put some time aside to start the migration.
I might also generate a Rust client here too as an excuse to get acquainted (I spend most of my time in Go these days).

**Comment by bermuda1023** [comment]
I would pay for a v4

**Comment by NathanBaulch** [comment]
Apologies for the radio silence on this folks - I've found some spare time and the v4 migration is progressing!
